### PR TITLE
cpr test: only return something for valid CPR numbers

### PR DESCRIPTION
**Pull request by danchr**
By 'valid' we mean the ones that represent a valid date of birth.

### CODE REVIEW

File: mora/integrations/serviceplatformen.py [hunk 1]
```diff
@@ -157,6 +158,9 @@ def get_citizen(cpr):
 
 
 def _get_citizen_stub(cpr):
+    if not util.is_cpr_number(cpr):
```
Comment (cmoesgaard) [inline mora/integrations/serviceplatformen.py]: Move this to `get_citizen` so we do this check regardless of running the stub or not.

File: mora/service/cpr.py [hunk 2]
```diff
@@ -48,12 +48,20 @@ def search_cpr():
...
-    else:
-        return '', 404
+    try:
+        sp_data = get_citizen(cpr)
+    except KeyError:
+        raise exceptions.NotFoundError(
+            'no such person found',
+            cpr=cpr,
+        )
+    except ValueError:
```
Comment (cmoesgaard) [inline mora/service/cpr.py]: Why catch and re-raise these exceptions? Wouldn't it make more sense to just raise the relevant errors in `get_citizen` directly?

Comment (danchr) [reply mora/service/cpr.py]: Layering? Figured the `integrations` package wasn't as MO-specific, and should raise plain exceptions, leaving the interpretation to the `cpr_lookup` service.

Comment (cmoesgaard) [reply mora/service/cpr.py]: I did intend it as something MO-specific, actually. The `integrations` package is to contain all of the adapters and logic for connecting to external systems - but I see your point. This way we could potentially extract this and reuse it somewhere else.

Comment (danchr) [reply mora/service/cpr.py]: Anyway, my thinking was that `KeyError` and `ValueError` talk of arguments to the current function, whereas the `exceptions` module is for shortcutting evaluation with an HTTP error — and that felt like something an integration shouldn't do.

**Pull request merged**